From b47616d69f00abef7966dd41db5bdb28ee340c6f Mon Sep 17 00:00:00 2001
From: Reza Alizadeh Majd <r.majd@pantherx.org>
Date: Sat, 19 Nov 2022 17:17:52 +0330
Subject: [PATCH] remove exe target for ios build

---
 src/CMakeLists.txt | 37 +++++++++++++++++++------------------
 1 file changed, 19 insertions(+), 18 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e3d58e9..dfe7d0f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -43,7 +43,7 @@ set(LIBRARY_SOURCES
   ${HEADERS}
   )
 
-set(PROGRAM "cmark_exe")
+# set(PROGRAM "cmark_exe")
 set(PROGRAM_SOURCES main.c)
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmark_version.h.in
@@ -52,19 +52,19 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmark_version.h.in
 include(GNUInstallDirs)
 include (GenerateExportHeader)
 
-add_executable(${PROGRAM} ${PROGRAM_SOURCES})
-cmark_add_compile_options(${PROGRAM})
-set_target_properties(${PROGRAM} PROPERTIES
-  OUTPUT_NAME "cmark")
+# add_executable(${PROGRAM} ${PROGRAM_SOURCES})
+# cmark_add_compile_options(${PROGRAM})
+# set_target_properties(${PROGRAM} PROPERTIES
+#   OUTPUT_NAME "cmark")
 
-if (CMARK_STATIC)
-  target_link_libraries(${PROGRAM} ${STATICLIBRARY})
-  # Disable the PUBLIC declarations when compiling the executable:
-  set_target_properties(${PROGRAM} PROPERTIES
-    COMPILE_FLAGS -DCMARK_STATIC_DEFINE)
-elseif (CMARK_SHARED)
-  target_link_libraries(${PROGRAM} ${LIBRARY})
-endif()
+# if (CMARK_STATIC)
+#   target_link_libraries(${PROGRAM} ${STATICLIBRARY})
+#   # Disable the PUBLIC declarations when compiling the executable:
+#   set_target_properties(${PROGRAM} PROPERTIES
+#     COMPILE_FLAGS -DCMARK_STATIC_DEFINE)
+# elseif (CMARK_SHARED)
+#   target_link_libraries(${PROGRAM} ${LIBRARY})
+# endif()
 
 # -fvisibility=hidden
 set(CMAKE_C_VISIBILITY_PRESET hidden)
@@ -121,17 +121,18 @@ if (CMARK_STATIC)
   list(APPEND CMARK_INSTALL ${STATICLIBRARY})
 endif()
 
-if (MSVC)
-  set_property(TARGET ${PROGRAM}
-    APPEND PROPERTY LINK_FLAGS /INCREMENTAL:NO)
-endif(MSVC)
+# if (MSVC)
+#   set_property(TARGET ${PROGRAM}
+#     APPEND PROPERTY LINK_FLAGS /INCREMENTAL:NO)
+# endif(MSVC)
 
 if(NOT MSVC OR CMAKE_HOST_SYSTEM_NAME STREQUAL Windows)
   set(CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_NO_WARNINGS ON)
   include(InstallRequiredSystemLibraries)
 endif()
 
-install(TARGETS ${PROGRAM} ${CMARK_INSTALL}
+# install(TARGETS ${PROGRAM} ${CMARK_INSTALL}
+install(TARGETS ${CMARK_INSTALL}
   EXPORT cmark-targets
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-- 
2.38.1

