From a3cecbf2cbe7b020da8b8f23276195c00e9f89be Mon Sep 17 00:00:00 2001
From: Reza Alizadeh Majd <r.majd@pantherx.org>
Date: Wed, 12 Oct 2022 14:39:53 +0330
Subject: [PATCH] add cmake build system

---
 CMakeLists.txt | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..6a0f51a
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.0)
+project(blurhash VERSION 0.1 LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+
+if (IOS)
+    set(LINK_TYPE STATIC)
+else ()
+    set(LINK_TYPE SHARED)
+endif ()
+
+add_library(${PROJECT_NAME} ${LINK_TYPE} blurhash.cpp)
+set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER "blurhash.hpp")
+
+install(TARGETS ${PROJECT_NAME}
+    LIBRARY DESTINATION lib
+    PUBLIC_HEADER DESTINATION include)
-- 
2.38.1

